### PR TITLE
Mulgen

### PR DESCRIPTION
**Pull request by jonlawlor**
as per #54

### CODE REVIEW

File: mat64/dense_arithmetic.go [hunk 1]
```diff
@@ -434,6 +434,168 @@ func (m *Dense) Mul(a, b Matrix) {
...
+	if a, ok := a.(Vectorer); ok {
+		if b, ok := b.(Vectorer); ok {
+			row := make([]float64, ac)
+			col := make([]float64, br)
+			if blasEngine == nil {
+				panic(ErrNoEngine)
+			}
+			if aTrans != blas.NoTrans {
+				if bTrans != blas.NoTrans {
+					for r := 0; r < ar; r++ {
```
Comment (jonlawlor) [inline mat64/dense_arithmetic.go]: The trouble is that would require a type assert to access the mat field, and we don't currently have a type that implements Vectorer but doesn't implement RawMatrixer.

**Comment by jonlawlor** [comment]
I've changed the code to use booleans, but I don't see how I can implement @btracey's suggestion.  Is it good to go without it?

**Comment by kortschak** [comment]
LGTM but again wait for @btracey.

File: mat64/dense_arithmetic.go (hunk 1 above)
Comment (kortschak) [inline mat64/dense_arithmetic.go]: You don't need to assert to get the mat field; w is a Dense.

Comment (jonlawlor) [inline mat64/dense_arithmetic.go]: Ah, yes you're right.

**Comment by jonlawlor** [comment]
OK, I've implemented that as well.  I haven't benchmarked it though.

**Comment by btracey** [comment]
LGTM.

**Pull request merged**